docs/specs/event.md

```diff
@@ -105,6 +108,10 @@ All fields are guaranteed that only backward-compatible changes would be made.
   - `portal`: The event originates from the management portal.
 - `preferred_languages`: User preferred languages which are inferred from the request. It is the value of `ui_locales` query parameter if it is provided, or the value of the `Accept-Language` header. It is an empty array when the event is not generated by the end user.
 - `language`: The locale which is derived based on user's preferred languages and app's languages config. The fallback value is the fallback language of the app.
+- `geo_location_code`: The (ISO 3166-1 alpha-2 code)[https://www.iban.com/country-codes] of the location derived from the ip address. `null` if the location cannot be determined by the ip address, for example, it is an internal ip address.
```

Comment:
Empty string if location cannot be determined

Reply:
Then maybe I keep it null. The user may want to look at the json to know which key he should check.